### PR TITLE
Remove old codebase and update dependencies

### PR DESCRIPTION
**Pull request by angelsl**
### Description

Remove old codebase, update all dependencies except TypeScript, migrate to Create React App/react-scripts (from the deprecated react-scripts-ts), and fix the resulting fallout.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Code quality improvements

### How to Test

Look at the CI test.

### Checklist

Please delete options that are not relevant.

- [x] I have tested this code
- [x] I have updated the documentation... probably

### CODE REVIEW

File: src/commons/sagas/BackendSaga.ts [hunk 1]
```diff
@@ -3,8 +3,7 @@
 import { SagaIterator } from 'redux-saga';
```
Comment (anthony-halim) [inline src/commons/sagas/BackendSaga.ts:3]: I think this file is not correctly merged from `remove-legacy` branch.

File: src/commons/utils/ActionsHelper.ts [hunk 2]
```diff
@@ -11,9 +11,6 @@ import * as SourcecastActions from '../../features/sourceRecorder/sourcecast/Sou
 import * as SourceRecorderActions from '../../features/sourceRecorder/SourceRecorderActions';
```
Comment (anthony-halim) [inline src/commons/utils/ActionsHelper.ts:11]: I think this file is not correctly merged from `remove-legacy` branch.

**Comment by angelsl** [comment]
Strange, the Travis build hanged. Happened twice already. Hmm.

**Comment by angelsl** [comment]
I got Yarn to rewrite `yarn.lock`, which removed about 200 duplicate dependencies.

**Pull request merged**